receiver/awscontainerinsightreceiver/internal/stores/utils.go

```diff
@@ -174,3 +178,92 @@ func refreshWithTimeout(parentContext context.Context, refresh func(), timeout t
 	<-ctx.Done()
 	cancel()
 }
+
+type RawContainerInsightsMetric struct {
+	// source of the metric for debugging merge conflict
+	ContainerName string
+	// key/value pairs that are typed and contain the metric (numerical) data
+	Fields map[string]any
+	// key/value string pairs that are used to identify the metrics
+	Tags   map[string]string
+	Logger *zap.Logger
+}
+
+var _ CIMetric = (*RawContainerInsightsMetric)(nil)
+
+func NewRawContainerInsightsMetric(mType string, logger *zap.Logger) *RawContainerInsightsMetric {
+	metric := &RawContainerInsightsMetric{
+		Fields: make(map[string]any),
+		Tags:   make(map[string]string),
+		Logger: logger,
+	}
+	metric.Tags[ci.MetricType] = mType
+	return metric
+}
+
+func NewRawContainerInsightsMetricWithData(mType string, fields map[string]any, tags map[string]string, logger *zap.Logger) *RawContainerInsightsMetric {
+	metric := NewRawContainerInsightsMetric(mType, logger)
+	metric.Fields = fields
+	metric.Tags = tags
+	return metric
+}
+
+func (c *RawContainerInsightsMetric) GetTags() map[string]string {
+	return c.Tags
+}
+
+func (c *RawContainerInsightsMetric) GetFields() map[string]any {
+	return c.Fields
+}
+
+func (c *RawContainerInsightsMetric) GetMetricType() string {
+	return c.Tags[ci.MetricType]
+}
+
+func (c *RawContainerInsightsMetric) AddTags(tags map[string]string) {
+	for k, v := range tags {
+		c.Tags[k] = v
+	}
+}
+
+func (c *RawContainerInsightsMetric) HasField(key string) bool {
+	return c.Fields[key] != nil
+}
+
+func (c *RawContainerInsightsMetric) AddField(key string, val any) {
+	c.Fields[key] = val
+}
+
+func (c *RawContainerInsightsMetric) GetField(key string) any {
+	return c.Fields[key]
+}
+
+func (c *RawContainerInsightsMetric) HasTag(key string) bool {
+	return c.Tags[key] != ""
+}
+
+func (c *RawContainerInsightsMetric) AddTag(key, val string) {
+	c.Tags[key] = val
+}
+
+func (c *RawContainerInsightsMetric) GetTag(key string) string {
+	return c.Tags[key]
+}
+
+func (c *RawContainerInsightsMetric) RemoveTag(key string) {
+	delete(c.Tags, key)
+}
+
+func (c *RawContainerInsightsMetric) Merge(src *RawContainerInsightsMetric) {
+	// If there is any conflict, keep the Fields with earlier timestamp
+	for k, v := range src.Fields {
+		if _, ok := c.Fields[k]; ok {
+			c.Logger.Debug(fmt.Sprintf("metric being merged has conflict in fields, src: %v, dest: %v \n", *src, *c))
+			c.Logger.Debug("metric being merged has conflict in fields", zap.String("src", src.ContainerName), zap.String("dest", c.ContainerName))
```

Comment:
Are there meant to be two of these? Seems like a merge that had a conflict.

Reply:
now sure how I missed this...